hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/access/AbstractAccessDelegate.java

```diff
@@ -108,11 +108,12 @@ public boolean putFromLoad(final Object key, final Object value, final long txTi
         return putFromLoad(key, value, txTimestamp, version);
     }
 
-    /**
-     * This is an asynchronous cache access strategy.
-     * NO-OP
-     */
     public void remove(final Object key) throws CacheException {
+        try {
+            cache.remove(key);
+        } catch (HazelcastException e) {
+            throw new CacheException("Operation timeout during remove operation from cache!", e);
+        }
```

Comment:
This was removed in https://github.com/hazelcast/hazelcast/pull/4110/files but seems to be essential (via `CollectionAction.evict` in LOCAL mode. 

Not sure why we never noticed this in DISTRIBUTED mode, perhaps with no backups we were always going to a remote node to populate collections.

@jhinch, Can you remember the reason for removing this in the first place?